utils/build-windows.bat

```diff
@@ -0,0 +1,287 @@
+:: See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+:: REQUIRED ENVIRONMENT VARIABLES
+:: This script requires to be executed inside one Visual Studio command line,
+:: in order for many of the tools and environment variables to be available.
+:: Additionally, it needs the following variables:
+:: - CMAKE_BUILD_TYPE: Kind of build: Release, RelWithDebInfo, Debug.
+:: - PYTHON_HOME: The Python installation directory.
+
+:: REQUIRED PERMISSIONS
+:: Practically, it is easier to be in the Adminstrators group to run the
+:: script, but it should be possible to execute as a normal user.
+:: The user will need permission to write files into the Windows SDK and the
+:: VisualC++ folder.
+
+@echo off
+
+setlocal enableextensions enabledelayedexpansion
+
+set "exitOnError=|| (exit /b)"
+set current_directory=%~dp0
+set current_directory=%current_directory:~0,-1%
+set source_root=%current_directory%\..\..
+
+cd %source_root%
+set source_root=%CD%
+set build_root=%source_root%\build
+set install_directory=%build_root%\Library\Developer\Toolchains\unknown-Asserts-development.xctoolchain\usr
+
+mkdir %build_root%
+
+call :clone_repositories %exitOnError%
+call :download_icu %exitOnError%
+:: TODO: Disabled until we need LLBuild/SwiftPM in this build script.
+:: call :download_sqlite3
+
+call :build_llvm %exitOnError%
+path %PATH%;%install_directory%\bin
+
+call :build_cmark %exitOnError%
+
+call :prepare_platform_modules %exitOnError%
+call :build_swift %exitOnError%
+
+call :build_lldb %exitOnError%
+
+path %source_root%\icu-63.1\bin64;%install_directory%\bin;%build_root%\swift\libdispatch-prefix\bin;%PATH%;%ProgramFiles%\Git\usr\bin
+call :test_swift %exitOnError%
+
+goto :end
+endlocal
+
+:clone_repositories
+:: Clones the repositories used by the Windows build.
+:: It supposes that the swift repository is already cloned by CI.
+:: It supposes the %CD% is the source root.
+setlocal enableextensions enabledelayedexpansion
+
+git clone --depth 1 --single-branch https://github.com/apple/swift-cmark cmark %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-clang clang %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-llvm llvm %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-compiler-rt compiler-rt %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-libdispatch %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-foundation %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-corelibs-xctest %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-lldb lldb %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-llbuild llbuild %exitOnError%
+git clone --depth 1 --single-branch https://github.com/apple/swift-package-manager %exitOnError%
```

Comment:
You are cloning this without the options that I made explicit in the docs.  You will fail random tests.

Reply:
You mean the `core.autocrlf` bits? If I understand correctly, the default behaviour on Windows will be `false`, which is set explicitly in the documentation, but shouldn’t be necessary. The `input` you have in Swift and SPM will turn CRLF into LF when checking out, but there should be no CRLF in the first place, right? Those bits are also not present in the Azure script, so I don’t know if they are really necessary.

PD: I was intending to use the `update-checkout` script for this step, but the changes to be compatible with Windows are still in review.

Reply:
This definitely is not the case in my experience.  The conversions that occur will result in failures for s-p-m and for swift.

Reply:
I have done two things. Seems that the Git Windows installer unattended installation sets the default to true. I added the global default to false in the script. I also removed a lot of checkouts that weren't being used for the moment (including s-p-m), which should make the checkouts leaner.

Looking around I don't see a way of checking out a git repository in Jenkins and specifying the configuration. I will setup the machine so the default is false (but including the extra step in the script will ensure that it keeps that way). The extra config you had in the docs (where the value is `input`), is only interesting if you commit to the repository, which will turn CRLF into LF on commit (but does nothing on checkout). Since the workers will not change the source or do commits, we should be fine.